packages/babel-plugin-transform-optional-chaining/src/index.js

```diff
@@ -0,0 +1,160 @@
+export default function ({ types: t }) {
+  const nilIdentifier = t.identifier("undefined");
```

Comment:
`t.unaryExpression('void', t.numericLiteral(0))` would be safer here; avoids any potential trouble with shadowing.